app/models/mongoidable/policy_relation.rb

```diff
@@ -17,7 +17,7 @@ def merged_policy_requirements
       merged_policy_requirements = Mongoidable::Abilities.new(mongoidable_identity, self)
       return merged_policy_requirements unless policy
 
-      policy_instance_abilities = policy.instance_abilities.clone
+      policy_instance_abilities = policy.instance_abilities.map(&:clone)
```

Comment:
So, what was happening here is that the old line was "cloning" the array.  Not the array elements.  This was roughly equivalent to:
```
policy_instance_abilities = policy.instance_abilities.map { |instance| instance }
```

So every `instance_ability` in both the original and the "clone" were the same memory pointer.

This wasn't a problem without caching because every Policy object was a new and separate object.

With caching, We are keeping the Policy in memory and sharing it.

Thus when we have the following situation:
```
  User 1 -> Organization A using Policy 1
  User 1 -> Organization B using Policy 1
```

Both `Policy 1` objects are the same in-memory object.

When we merge the organization policy stuff for `current_abilities` we're actually changing that policies in-memory copy so `Policy 1` now becomes "hard coded" for that one specific organization, and everything that uses `Policy 1` gets that hard coded value.

Fixing this fixes our trouncing memory we shouldn't be changing, and puts us back in the race with caching probably being a good idea.

Reply:
Can you write a test to prove this?